pkg/domain/assetFetcher.go

```diff
@@ -7,4 +7,5 @@ import (
 // AssetFetcher represents the interface you can use to fetch scanned assets for a given site
 type AssetFetcher interface {
 	FetchAssets(ctx context.Context, siteID string) (<-chan AssetEvent, <-chan error)
+	CheckDependencies(ctx context.Context) error
```

Comment:
Question: why are you adding `CheckDependencies` to the AssetFetcher interface? That feels very, very wrong.